src/dc_tools.rs

```diff
@@ -642,13 +642,6 @@ impl rusqlite::types::ToSql for EmailAddress {
     }
 }
 
-/// Utility to check if a in the binary represantion of listflags
-/// the bit at position bitindex is 1.
-pub(crate) fn listflags_has(listflags: u32, bitindex: usize) -> bool {
```

Comment:
iirc, this was introduced at some point to make things more readable, not sure if that is fulfilled and needed this way - also as on similar placed no special function is used.

but i do not understand the comment "the implementation worked correctly only for bitindex=1 and bitindex=2." from above - there is even a test for other bitindex.

Reply:
But it's not a bit *index* or bit *position*. It is a bitmask. If bit index is `3`, the bit mask should be `8`. At the very least the type of `bitindex` should be `u32` too, same as the type of value masked.

Reply:
true! i have never read the documentation that way and did not thought much about parameter names. this type of confusion is one more reason to do the bit-thingies directly, indeed.